# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/Relation.h
# Commit: 8d31c0604efe
# Full Hash: 8d31c0604efef2015995df32766d38dd4dd45ca2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 1: Support base Accessible in Relation. r=morgan
#   
#   Previously, Relation::Next returned a LocalAccessible.
#   Now, it returns an Accessible so that it can support RemoteAccessible as well.
#   Relation::LocalNext has been added as a convenience to return a LocalAccessible when the caller is certain that only LocalAccessible is relevant.
# ==============================================================================

diff -r af7f9589f48c -r 8d31c0604efe accessible/base/Relation.h
--- a/accessible/base/Relation.h	Thu Aug 04 02:47:34 2022 +0300
+++ b/accessible/base/Relation.h	Wed Aug 03 23:58:50 2022 +0000
@@ -74,7 +74,7 @@
   /**
    * compute and return the next related accessible.
    */
-  inline LocalAccessible* Next() {
+  inline Accessible* Next() {
     Accessible* target = nullptr;
 
     while (mFirstIter && !(target = mFirstIter->Next())) {
@@ -83,7 +83,12 @@
 
     if (!mFirstIter) mLastIter = nullptr;
 
-    return target ? target->AsLocal() : nullptr;
+    return target;
+  }
+
+  inline LocalAccessible* LocalNext() {
+    Accessible* next = Next();
+    return next ? next->AsLocal() : nullptr;
   }
 
  private: